Taurine/app/ViewController.swift

```diff
@@ -262,7 +262,7 @@ class ViewController: UIViewController, ElectraUI {
                 case .kfdPhysPuppet:
                     print("Selecting kfd [physpuppet] for iOS 14.0 - 14.8.1")
                     LogStream.shared.pause()
-                    let ret = do_kopen(0x800, 0x0, 0x2, 0x2)
+                    let ret = do_kopen(0x20000, 0x0, 0x2, 0x2)
```

Comment:
```suggestion
                    var systemInfo = utsname()
                    uname(&systemInfo)
                    let machineMirror = Mirror(reflecting: systemInfo.machine)
                    let deviceModel = kernelMirror.children.reduce("") { identifier, element in
                        guard let value = element.value as? Int8, value != 0 else { return identifier }
                        return identifier + String(UnicodeScalar(UInt8(value)))
                    }
                    let ret;
                    if (deviceModel.contains("iPad13") && deviceModel != "iPad13,1" && deviceModel != "iPad13,2") {
                        // M1 iPad Pro
                        ret = do_kopen(0x20000, 0x0, 0x2, 0x2)
                    } else {
                        ret = do_kopen(0x800, 0x0, 0x2, 0x2)
                    }
```

detect M1.

Sorry I don't know how to make reviews for multiple files so making multiple reviews for them :P